### ISSUE TITLE
descriptor_parse: crash with `pk(xprv.../*)`

### ISSUE THREAD

**Issue by brunoerg** [brunoerg/bitcoinfuzz#70]
Input: `pk(xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/*)`

Bitcoin Core successfully parses it and rust-miniscript returned error "Public keys must be 64/66/130 characters in size"/"pubkey string should be 66 or 130 digits long, got: 113".

**Comment by apoelstra** [comment]
Neat! Can you move/copy this to rust-miniscript. I think this is a gap in our definition of `DescriptorPublicKey`.

**Comment by brunoerg** [comment]
> Neat! Can you move/copy this to rust-miniscript. I think this is a gap in our definition of DescriptorPublicKey.

Yes! Done. https://github.com/rust-bitcoin/rust-miniscript/issues/785

**Comment by trevarj** [comment]
I could hack this to work by adding a case to `FromStr for DescriptorPublicKey` that checks if it's a `xprv` key and then just derive the `xpub`, but I'm not sure if that's what @apoelstra has in mind.

I think someone might also expect parsing as a `Descriptor<DescriptorSecretKey>` (or even `Descriptor<DescriptorXKey<Xprv>>` to work, but they don't.

**Comment by brunoerg** [comment]
Closing here since there is an issue on rust-miniscript for this.
